diesel/src/connection/transaction_manager.rs

```diff
@@ -110,12 +110,21 @@ where
         )
     }
 
+    /// If the transaction fails to commit due to a `SerializationFailure` or a
+    /// `ReadOnlyTransaction` rollback will be attempted. If it succeeds,
+    /// original error will be returned. If it fails, rollback error will be returned.
     fn commit_transaction(&self, conn: &Conn) -> QueryResult<()> {
         let transaction_depth = self.transaction_depth.get();
         self.change_transaction_depth(
             -1,
             if transaction_depth <= 1 {
-                conn.batch_execute("COMMIT")
+                conn.batch_execute("COMMIT").or_else(|err| match err {
+                    Error::DatabaseError(DatabaseErrorKind::SerializationFailure, _)
```

Comment:
Maybe add a small note here why we have choosen to only handle those two error kinds? It should probably also state that it is fine to add other cases if it turns out to be a problem later.

Reply:
done